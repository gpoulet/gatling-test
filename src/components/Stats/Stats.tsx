import React, { FC } from "react";
import { Pie } from "@reactchartjs/react-chart.js";
import { ChartOptions } from "chart.js";

export type StatsProps = {
  numberOfUserLivingInAnApt: number;
  numberOfUserLivingInASuite: number;
};

let options: ChartOptions = {
  legend: {
    position: "bottom",
  },
};

const Stats: FC<StatsProps> = ({
  numberOfUserLivingInAnApt,
  numberOfUserLivingInASuite,
}) => {
  const data = {
    labels: [
      "Number of user living in an apt",
      "Number of user living in a suite",
    ],
    datasets: [
      {
        data: [numberOfUserLivingInAnApt, numberOfUserLivingInASuite],
        backgroundColor: ["#36A2EB", "#FFCE56"],
        hoverBackgroundColor: ["#36A2EB", "#FFCE56"],
      },
    ],
  };
  return (
    <div className="stats">
      <Pie data={data} options={options} type="pie" />
    </div>
  );
};

export default Stats;
