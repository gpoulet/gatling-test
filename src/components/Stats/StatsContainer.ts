import {connect} from "react-redux";
import Stats from "./Stats";
import {getNumberOfUserLivingInAnApt, getnumberOfUserLivingInASuite} from "../../selectors";
import {RootState} from "../../store";

const mapStateToProps = (state: RootState) => ({
    numberOfUserLivingInAnApt: getNumberOfUserLivingInAnApt(state),
    numberOfUserLivingInASuite: getnumberOfUserLivingInASuite(state)
})

export default connect(mapStateToProps)(Stats)