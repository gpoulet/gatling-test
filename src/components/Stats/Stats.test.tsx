import React from 'react';
import renderer from 'react-test-renderer';
import Stats from './Stats';

it('Stats renders correctly', () => {
  const tree = renderer
    .create(<Stats numberOfUserLivingInAnApt={4} numberOfUserLivingInASuite={8} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});