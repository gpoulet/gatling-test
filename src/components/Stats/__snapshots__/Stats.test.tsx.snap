// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats renders correctly 1`] = `
<div
  className="stats"
>
  <canvas
    data-testid="canvas"
    height={150}
    onClick={[Function]}
    width={300}
  />
</div>
`;
